Gems/EMotionFX/Code/EMotionFX/Source/AnimGraphNode.cpp

```diff
@@ -2537,4 +2571,53 @@ namespace EMotionFX
             ->Attribute(AZ_CRC("AnimGraph", 0x0d53d4b3), &AnimGraphNode::GetAnimGraph)
         ;
     }
+
+
+#if defined(EMFX_ANIMGRAPH_PROFILER_ENABLED)
+    void AnimGraphNode::ClearProfileTimers(AnimGraphInstance* animGraphInstance)
+    {
+        AnimGraphNodeData* uniqueData = FindOrCreateUniqueNodeData(animGraphInstance);
+        uniqueData->ClearUpdateTimes();
+    }
+
+    AZStd::chrono::microseconds AnimGraphNode::GetTotalUpdateTime(AnimGraphInstance* animGraphInstance) const
+    {
+        AnimGraphNodeData* uniqueData = FindOrCreateUniqueNodeData(animGraphInstance);
+        return AZStd::chrono::duration_cast<AZStd::chrono::microseconds>(uniqueData->GetTotalUpdateTime());
+    }
+
+    AZStd::chrono::microseconds AnimGraphNode::GetUpdateTime(AnimGraphInstance* animGraphInstance) const
+    {
+        AnimGraphNodeData* uniqueData = FindOrCreateUniqueNodeData(animGraphInstance);
+        return AZStd::chrono::duration_cast<AZStd::chrono::microseconds>(uniqueData->GetTotalUpdateTime() - uniqueData->GetInputNodesUpdateTime());
+    }
+
+    AnimGraphNode::ProfileSection::ProfileSection(AnimGraphNode* node, AnimGraphInstance* animGraphInstance, ProfileMode mode, bool incomingNode)
+        : m_node(node)
+        , m_animGraphInstance(animGraphInstance)
+        , m_profileMode(mode)
+        , m_isIncomingNode(incomingNode)
+        , m_startPoint(AZStd::chrono::system_clock::now())
+    {
+    }
+
+    AnimGraphNode::ProfileSection::~ProfileSection()
+    {
+        if (m_node->GetProfileMode() & static_cast<AZ::u8>(m_profileMode))
+        {
+            AZStd::chrono::nanoseconds duration = AZStd::chrono::system_clock::now() - m_startPoint;
+            if (AnimGraphNodeData* pUniqueData = m_node->FindOrCreateUniqueNodeData(m_animGraphInstance))
```

Comment:
nit/standards: Is it possible to do a pass on the naming to remove the hungarian notation we don't usually use (so this just becomes `uniqueData` (not `pUniqueData`)

Reply:
good call